src/webhook/validation/oneagent_test.go

```diff
@@ -290,3 +292,242 @@ func TestImageFieldSetWithoutCSIFlag(t *testing.T) {
+		})
+	}
+}
+
+type validOneAgentVolumeStorageReadOnlyModeSettingsTestcase struct {
+	testName             string
+	dynakubeFactory      func(featureFlag string, featureFlagValue bool, oaEnvVar string, oaEnvValue bool) *dynatracev1beta1.DynaKube
+	featureFlag          string
+	featureValue         bool
+	oaVolumeStorageVar   string
+	oaVolumeStorageValue bool
+	acceptedWarningCount int
+}
+
+func TestOneAgentVolumeStorageReadOnlyModeNoConflict(t *testing.T) {
+	testcases := []validOneAgentVolumeStorageReadOnlyModeSettingsTestcase{
+		{"enabled OneAgent volume storage by env var with OneAgent read only mode disabled by feature flag in cloudNative",
+			createCloudNativeFullstackDynaKube,
+			dynatracev1beta1.AnnotationFeatureReadOnlyOneAgent, false,
+			oneAgentVolumeStorageEnvVar, true,
+			0},
+		{"enabled OneAgent volume storage by env var with OneAgent read only mode disabled by feature flag in hostMonitoring",
+			createHostMonitoringDynaKube,
+			dynatracev1beta1.AnnotationFeatureReadOnlyOneAgent, false,
+			oneAgentVolumeStorageEnvVar, true,
+			0},
+		{"enabled OneAgent volume storage with OneAgent read only mode disabled by deprecated feature flag in cloudNative",
+			createCloudNativeFullstackDynaKube,
+			dynatracev1beta1.AnnotationFeatureDisableReadOnlyOneAgent, true,
+			oneAgentVolumeStorageEnvVar, true,
+			1},
+		{"enabled OneAgent volume storage with OneAgent read only mode disabled by deprecated feature flag in hostMonitoring",
+			createHostMonitoringDynaKube,
+			dynatracev1beta1.AnnotationFeatureDisableReadOnlyOneAgent, true,
+			oneAgentVolumeStorageEnvVar, true,
+			1},
+		{"OneAgent volume storage default with OneAgent read only mode enabled by feature flag in cloudNative",
+			createCloudNativeFullstackDynaKube,
+			dynatracev1beta1.AnnotationFeatureReadOnlyOneAgent, true,
+			"", true,
+			0},
+		{"OneAgent volume storage default with OneAgent read only mode enabled by feature flag in hostMonitoring",
+			createHostMonitoringDynaKube,
+			dynatracev1beta1.AnnotationFeatureReadOnlyOneAgent, true,
+			"", true,
+			0},
+		{"OneAgent volume storage default with OneAgent read only mode enabled by default in cloudNative",
+			createCloudNativeFullstackDynaKube,
+			"", true,
+			"", true,
+			0},
+		{"OneAgent volume storage default with OneAgent read only mode enabled by default in hostMonitoring",
+			createHostMonitoringDynaKube,
+			"", true,
+			"", true,
+			0},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.testName, func(t *testing.T) {
+			assertAllowedResponseWithWarnings(t,
+				tc.acceptedWarningCount,
+				tc.dynakubeFactory(tc.featureFlag, tc.featureValue, tc.oaVolumeStorageVar, tc.oaVolumeStorageValue),
+				&defaultCSIDaemonSet)
+		})
+	}
+}
+
+func TestReadOnlyModeClassicFullstackWarning(t *testing.T) {
+	testcases := []validOneAgentVolumeStorageReadOnlyModeSettingsTestcase{
+		{"OneAgent volume storage disabled by env var and OneAgent read-only mode enabled by feature flag in classicFullstack",
```

Comment:
the env variable is irrelevant for classic fullstack. we provide a warning when the feature flag is used, no matter if the env variable is set or not.

Reply:
True, but this tests are to make sure that we only get warnings and not any of the newly introduced errors in classicFullstack. So they are not only to test the "warning-feature" but also to make sure the "validation-feature" doesn't run havoc.